app/src/main/java/org/p2p/wallet/jupiter/statemanager/SwapStateManager.kt

```diff
@@ -308,11 +308,11 @@ class SwapStateManager(
     }
 
     private fun observeUserTokens() {
-        homeLocalRepository.getTokensFlow()
-            .mapLatest { userTokens ->
-                userTokensChangeHandler.handleUserTokensChange(state.value, userTokens)
+        tokenServiceCoordinator.observeUserTokens()
+            .filterIsInstance<UserTokensState.Loaded>()
+            .onEach { userTokensState ->
```

Comment:
try to keep mapLatest here and update new state with onEach

Reply:
okay